upgrade-product-recalculations.md

```diff
@@ -0,0 +1,586 @@
+# Upgrade Product Recalculations
+
+## General changes that have to be known
+
+This article covers the product recalculations, which means recalculation of the product visibility, selling denial, and export to Elasticsearch.
+
+Previous implementation was not unified and therefore may be sometimes tricky – for example, visibility had to be calculated before export to Elasticsearch; otherwise the product may be exported with wrong visibility.
```

Comment:
```suggestion
The previous implementation was not unified and, therefore, may sometimes be tricky. For example, visibility had to be calculated before exporting to Elasticsearch; otherwise, the product may be exported with the wrong visibility.
```